spec/controllers/admin/search/purchases_controller_spec.rb

```diff
@@ -12,18 +13,36 @@
     sign_in create(:admin_user)
   end
 
-  describe "#purchases" do
+  describe "#index" do
     let!(:email) { "user@example.com" }
     let(:ip_v4) { "203.0.113.42" }
 
+    it "returns successful response with Inertia page data" do
+      get :index, params: { query: email }
+
+      expect(response).to be_successful
+      expect(inertia.component).to eq("Admin/Search/Purchases/Index")
+    end
+
+    it "returns JSON response when requested" do
```

Comment:
Might be nice to add a spec for pagination as well. I think a couple of the other admin controllers have them

Reply:
Added to this spec!